src/collectors/dmi_collector.rs

```diff
@@ -0,0 +1,282 @@
+/*
+ * TODO:
+ * 1. Implement Error checking.
+ * 2. Check netbox api documentation for correct information type.
+ * 3. REMOVE DEBUG PRINT STATEMENTS.
+ * */
+
+use std::{str::Split, process::{Command, Output}};
+use super::util::{find_table, split_output};
+
+// Fully collected DMI information
+#[derive(Debug)]
+pub struct DmiInformation {
+   system_information: SystemInformation,
+   chassis_information: ChassisInformation,
+   cpu_information: CpuInformation,
+}
+
+// General system information
+#[derive(Debug)]
+pub struct SystemInformation {
+    vendor: String,
+    model: String,
+    uuid: String,
+    serial: String,
+}
+
+#[derive(Debug)]
+pub struct ChassisInformation {
+    asset: String,
+}
+
+// CPU Information
+#[derive(Debug)]
+pub struct CpuInformation {
+    version: String,
+    core_count: String,
+    cores_enabled: String,
+    thread_count: String,
+    max_speed: String,
+    voltage: String,
+    status: String,
+}
+
+
+fn execute_dmidecode(dmidecode_table: i32) -> String {
+    /*
+     * Collect DMI information from System.
+     *
+     * This function executes the dmidecode command for the table type provided.
+     */
+    let output: Output = Command::new("sudo")
+        .arg("dmidecode")
+        .arg("-t")
+        .arg(dmidecode_table.to_string())
+        .output()
+        .expect("Failed to execute command");
+
+    // Read the output of the command
+    return String::from_utf8_lossy(&output.stdout).to_string();
```

Comment:
I think this should be a from_utf8 (https://doc.rust-lang.org/stable/std/string/struct.String.html#method.from_utf8) with error handling

Reply:
This does not seem to work due to the types `from_utf8` expects. The execution of `dmidecode` returns an output object which I am trying to force into a string format. I will look into it however.